cmd/gardenlet/app/app.go

```diff
@@ -672,6 +679,30 @@ func getSecretsToRecreate(ctx context.Context, c client.Client, namespacesInDele
 	return secretsToRecreate, nil
 }
 
+func cleanupShootCoreManagedResource(ctx context.Context, seedClient client.Client) error {
+	managedResourceList := &metav1.PartialObjectMetadataList{}
+	managedResourceList.SetGroupVersionKind(resourcesv1alpha1.SchemeGroupVersion.WithKind("ManagedResourceList"))
+	if err := seedClient.List(ctx, managedResourceList); err != nil {
+		if meta.IsNoMatchError(err) {
+			return nil
+		}
+		return err
+	}
+
+	var taskFns []flow.TaskFn
+	for _, managedResource := range managedResourceList.Items {
+		if managedResource.GetName() == "shoot-core" {
+			mr := managedResource
+
+			taskFns = append(taskFns, func(ctx context.Context) error {
+				return managedresources.DeleteForShoot(ctx, seedClient, mr.Namespace, "shoot-core")
+			})
+		}
+	}
+
+	return flow.Parallel(taskFns...)(ctx)
+}
```

Comment:
Instead of this would it be simple to delete every time and Ignore if not found? As of now we are listing all MR again and again that's an expensive call.

Reply:
WDYM by again and again? It's only done once at the startup of gardenlet.